### DISCUSSION TITLE
Support adding "radios" as playlist

### DISCUSSION THREAD

**Discussion by cu** [navidrome/navidrome#2753] [Ideas]
Most audio streams utilize well-known playlist URLs which point to dynamically-generated playlists. This allows the stream operator to direct listeners to specific streaming servers, for load balancing, redundancy, or to accommodate infrastructure maintenance with minimal impact to listeners.

As an example, if you go to somafm.com and click a link for a stream called The Trip, you will get a URL like this: https://somafm.com/thetrip.pls

The contents of this file are a playlist of streams:

```
[playlist]
numberofentries=5
File1=https://ice6.somafm.com/thetrip-128-mp3
Title1=SomaFM: The Trip (#1): Progressive house / trance. Tip top tunes.
Length1=-1
File2=https://ice4.somafm.com/thetrip-128-mp3
Title2=SomaFM: The Trip (#2): Progressive house / trance. Tip top tunes.
Length2=-1
File3=https://ice2.somafm.com/thetrip-128-mp3
Title3=SomaFM: The Trip (#3): Progressive house / trance. Tip top tunes.
Length3=-1
File4=https://ice1.somafm.com/thetrip-128-mp3
Title4=SomaFM: The Trip (#4): Progressive house / trance. Tip top tunes.
Length4=-1
File5=https://ice5.somafm.com/thetrip-128-mp3
Title5=SomaFM: The Trip (#5): Progressive house / trance. Tip top tunes.
Length5=-1
Version=2
```

Depending on when you fetch this playlist, the order of the audio streams will likely be different.

Because I am a technical user, I know that Navidrome currently does not support stream playlist URLs, only audio stream URLs. So I know to download the playlist, pick an audio stream server URL from the playlist, and paste that into Navidrome. (And hope that that particular stream URL is not taken out of service in the future.) 

My request here is to support adding streams via a playlist URL as well as the direct URL to the audio stream. When a stream is added as a playlist, only the playlist URL should be stored in the database, such that the playlist is downloaded and evaluated to find an audio stream to play, every time the user elects to play that "radio station" in the UI.

The chief advantages here are:

* More user-friendly: Non-technical users may not know the difference between an audio stream and a playlist. Supporting playlists natively in the application means they don't have to know or care about the difference.
* Better behavior: stream operators may add/remove audio streams and servers to their playlist(s) for various reasons. This makes Navidrome "play nicely" with stream operators who may be attempting to prioritize and/or load-balance certain audio stream servers.

This has come up (indirectly) at least once before: https://github.com/navidrome/navidrome/discussions/2455

**Comment by jm355** [comment]
+1 for this. In addition, selecting a random stream from the list is important for load balancing, and in case one of the streams is down. 

Another nice feature that might be better off as a separate discussion: allowing to set a mobile/metered stream url. To use somafm as the example again, they offer a variety of streaming qualities. I like to use the [aac 128k version](https://somafm.com/groovesalad130.pls) on unmetered wifi, and the [aac 32k version](https://somafm.com/groovesalad32.pls) on mobile or metered wifi. Symfonium supports setting a mobile stream for radios, and I think it'd be nice if navidrome did too
